src/Compilers/Core/Portable/Operations/InstanceReferenceKind.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    /// <summary>
+    /// Kind of reference for an <see cref="IInstanceReferenceOperation"/>.
+    /// </summary>
+    public enum InstanceReferenceKind
+    {
+        /// <summary>
+        /// Reference to the containing type. Used for <code>this</code> and <code>base</code> in C# code, and <code>This</code>,
+        /// <code>MyClass</code>, <code>MyBase</code> in VB code.
+        /// </summary>
+        ContainingTypeInstance,
+        /// <summary>
+        /// Reference to a member of the object being initialized in a C# object or collection initializer, or a VB With statement.
```

Comment:
>Reference to a member [](start = 12, length = 21)

This doesn't sound right. I don't think IInstanceReferenceOperation ever references a member.